src/dispatch/plugins/dispatch_slack/service.py

```diff
@@ -1,43 +1,49 @@
 import functools
-import inspect
 import logging
 import time
-from typing import Any, Dict, List, Optional
 
-import slack_sdk
+from typing import Dict, List, Optional, Union
+
 from blockkit import Message, Section
-from slack_sdk.web.async_client import AsyncWebClient
+from slack_sdk.errors import SlackApiError
+from slack_sdk.web.client import WebClient
+from slack_sdk.web.slack_response import SlackResponse
+
 from tenacity import TryAgain, retry, retry_if_exception_type, stop_after_attempt
 
 from .config import SlackConversationConfiguration
 
-log = logging.getLogger(__name__)
+# List of Slack endpoints that require HTTP GET method
+SLACK_GET_ENDPOINTS = [
+    "conversations.history",
+    "conversations.info",
+    "users.conversations",
+    "users.info",
+    "users.lookupByEmail",
+    "users.profile.get",
+]
 
+Conversation = dict[str, str]
 
-def fullname(o):
-    module = inspect.getmodule(o)
-    return f"{module.__name__}.{o.__qualname__}"
+log = logging.getLogger(__name__)
 
 
-def create_slack_client(config: SlackConversationConfiguration, run_async: bool = False):
+def create_slack_client(config: SlackConversationConfiguration) -> WebClient:
```

Comment:
I wonder if we could use the existing client attached to the Bolt `app` defined in `bolt.py`

https://slack.dev/bolt-python/concepts#web-api

Reply:
Good call out. I'll look into it.